src/charm.py

```diff
@@ -174,6 +183,7 @@ def _generate_root_certificate(self) -> None:
             secret = self.model.get_secret(label=CA_CERTIFICATES_SECRET_LABEL)
             secret.set_content(content=secret_content)
         else:
+            self._push_ca_cert_to_container(ca_certificate.decode())
```

Comment:
Shouldn't we do this in both cases (outside the if)?  Otherwise we won't update it on the workload if the certificate changes.

Reply:
Yes, good point. Fixed.